cmd/podman/machine/platform_windows.go

```diff
@@ -1,10 +1,32 @@
 package machine
 
 import (
+	"os"
+	"strings"
+
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/wsl"
+	"github.com/sirupsen/logrus"
 )
 
-func GetSystemDefaultProvider() machine.VirtProvider {
-	return wsl.GetWSLProvider()
+func GetSystemProvider(name string) machine.VirtProvider {
+	provider := name
+	providerOverride, found := os.LookupEnv("CONTAINERS_MACHINE_PROVIDER")
+	if found {
+		provider = providerOverride
+	}
+	provider = strings.ToUpper(strings.TrimSpace(provider))
+	switch provider {
+	case "WSL":
```

Comment:
I'm sitting on a PR where I strong typed all providers.  My preference is we do something like that for sure.  Should I yank that out of my PR and submit it?

Reply:
> I'm sitting on a PR where I strong typed all providers

I have no objections about this. Strong typed is often a better option. If this will be split between common and podman repos it would make things slightly more difficult for external contributors, but I believe that typed version is still better.

Reply:
Reworked into typed version. I really don't like that I had to implement nullability with a pointer, because there is no Undefined value in the enum. Still thinking about reworking API into 2 separate functions - one taking enum parameter and another one w/o any parameters and then handling nullability in the private function.